test/metabase/sync/analyze/fingerprint/fingerprinters_test.clj

```diff
@@ -100,3 +96,32 @@
          (transduce identity
                     (fingerprinter (field/map->FieldInstance {:base_type :type/Text}))
                     ["metabase" "more" "like" "metabae" "[1, 2, 3]"]))))
+
+(deftest fingerprints-in-db-test
+  (mt/test-drivers (mt/normal-drivers)
+    (testing "Fingerprints should actually get saved with the correct values"
+      (testing "Text fingerprints"
+        (is (schema= {:global {:distinct-count (s/eq 100)
+                               :nil%           (s/eq 0.0)}
+                      :type   {:type/Text {:percent-json   (s/eq 0.0)
+                                           :percent-url    (s/eq 0.0)
+                                           :percent-email  (s/eq 0.0)
+                                           :average-length (s/pred #(< 15 % 16) "between 15 and 16")}}}
+                     (db/select-one-field :fingerprint Field :id (mt/id :venues :name)))))
+      (testing "date fingerprints"
+        (is (= {:global {:distinct-count 618
+                         :nil%           0.0}
+                :type   {:type/DateTime {:earliest "2013-01-03"
+                                         :latest   "2015-12-29"}}}
+               (db/select-one-field :fingerprint Field :id (mt/id :checkins :date)))))
+
+      (testing "number fingerprints"
+        (is (schema= {:global {:distinct-count (s/eq 4)
```

Comment:
We also have some rounding testing facilities which might be more clear than using schema. See fingerprint rounding fns in `metabase.test.util`

Reply:
ahhh okay sweet.